libs/net/dal/Services/NotificationService.cs

```diff
@@ -95,5 +192,43 @@ public async Task<int> Unsubscribe(int userId)
         });
         return saveChanges ? await Context.SaveChangesAsync() : await Task.FromResult(0);
     }
+
+    /// <summary>
+    /// Find the last notification instance created for the specified 'notificationId' and 'ownerId'.
+    /// </summary>
+    /// <param name="notificationId"></param>
+    /// <param name="ownerId"></param>
+    /// <returns></returns>
+    private NotificationInstance? GetPreviousNotificationInstance(int notificationId, int? ownerId)
+    {
+        return this.Context.NotificationInstances
+            .OrderByDescending(i => i.Id)
+            .Where(i => i.NotificationId == notificationId &&
+                i.OwnerId == ownerId)
+            .FirstOrDefault();
+    }
+
+    /// <summary>
+    /// Make a request to Elasticsearch to find content for the specified 'notification'.
+    /// </summary>
+    /// <param name="notification"></param>
+    /// <param name="requestorId"></param>
+    /// <returns></returns>
+    public async Task<Elastic.Models.SearchResultModel<API.Areas.Services.Models.Content.ContentModel>> FindContentWithElasticsearchAsync(Notification notification, int? requestorId)
```

Comment:
The main issue I see with this implementation is that the first time the notification runs it will get a lot of content...  I'll need to create a story to handle this.